shaarli_client/v1.py

```diff
@@ -0,0 +1,30 @@
+"""Shaarli REST API v1 client"""
+import calendar
+import time
+
+import requests
+from requests_jwt import JWTAuth
+
+
+class ShaarliV1Client:
+    """Shaarli REST API v1 client"""
+
+    # pylint: disable=too-few-public-methods
+
+    def __init__(self, uri, secret):
+        """Client constructor"""
+        self.uri = uri
+        self.secret = secret
+        self.version = 1
+
+    def _request(self, method, endpoint, params):
+        """Send an HTTP request to this instance"""
+        auth = JWTAuth(self.secret, alg='HS512', header_format='Bearer %s')
+        auth.add_field('iat', lambda req: calendar.timegm(time.gmtime()))
```

Comment:
`lambda` expressions are convenient to define short-lived/local functions, and are used here to pass a _callable_ to [`auth.add_field`](https://requests-jwt.readthedocs.io/en/latest/#requests_jwt.JWTAuth.add_field) so it can be _called_ when needed; the same expression without using `lambda` would amount to defining and passing a function object:

```python
def utc_epoch():
    """Custom wrapper"""
    return calendar.timegm(time.gmtime())

[...]
    auth.add_field('iat', utc_epoch)
```

More on lambda expressions and callables:
- https://docs.python.org/3.5/tutorial/controlflow.html#lambda-expressions
- https://stackoverflow.com/questions/890128/why-are-python-lambdas-useful
- https://stackoverflow.com/questions/111234/what-is-a-callable-in-python#111255